src/escapeField.spec.ts

```diff
@@ -15,3 +16,113 @@ describe("escapeField function", () => {
     );
   });
 });
+
+describe("escapeField property-based tests", () => {
+  it("should quote fields with special characters or delimiters", () => {
+    fc.assert(
+      fc.property(fc.array(fc.string()), (row) => {
+        for (const field of row) {
+          const escaped = escapeField(field);
+          if (/[\n\r",]/.test(field)) {
+            expect(escaped.startsWith('"') && escaped.endsWith('"')).toBe(true);
+          }
+        }
+      }),
+    );
+  });
+
+  it("should always quote when quote=true", () => {
+    fc.assert(
+      fc.property(fc.array(fc.string()), (row) => {
+        for (const field of row) {
+          const escaped = escapeField(field, { quote: true });
+          expect(escaped.startsWith('"') && escaped.endsWith('"')).toBe(true);
+        }
+      }),
+    );
+  });
+
+  it("should correctly escape internal quotation marks", () => {
+    fc.assert(
+      fc.property(fc.array(fc.string()), (row) => {
+        for (const field of row) {
+          const escaped = escapeField(field, { quote: true });
+          if (field.includes('"')) {
+            const inner = escaped.slice(1, -1);
+            const originalCount = field.split('"').length - 1;
+            const doubledCount = (inner.match(/""/g) || []).length;
+            expect(doubledCount).toBe(originalCount);
+          }
+        }
+      }),
+    );
+  });
+
+  it("should work with custom quotation characters", () => {
+    fc.assert(
+      fc.property(
+        fc.array(fc.string()),
+        fc.constantFrom("'", "`", "$", "cc"),
```

Comment:
This applies to other parts as well, but rather than using your own definitions, I think it's better to use `FC.text()`, `FC.quotation()`, etc., which are already implemented in `src/__tests__/helper.ts`.

Also, the essence of the test is at the field level, so there's no need to do it at the row level using `fc.array(fc.string())`. I think `fc.property(FC.text(), FC.quotation())` or similar would be better.